be/src/storage/delta_writer.h

```diff
@@ -65,7 +65,9 @@ struct DeltaWriterOptions {
     ReplicaState replica_state;
     bool miss_auto_increment_column = false;
     PartialUpdateMode partial_update_mode = PartialUpdateMode::UNKNOWN_MODE;
-    POlapTableSchemaParam ptable_schema_param;
+    // `ptable_schema_param` is valid during initialization.
+    // And it may become invalid after initialization, so please do not access it afterward.
+    const POlapTableSchemaParam* ptable_schema_param = nullptr;
```

Comment:
If this will be invalid after unit, why not make it a function variable in unit rather than store it in member variable. 
And it will be better to set it to null after unit then it will cause crash rather than wild pointer issues

Reply:
> If this will be invalid after unit, why not make it a function variable in unit rather than store it in member variable. And it will be better to set it to null after unit then it will cause crash rather than wild pointer issues

Actually this point is valid during data ingestion right now, but I worry that someone else might prematurely release the corresponding memory when making changes, so I added some comments as a warning. And we only need to access this memory during initialization. I reset it to nullptr after init and modified some comments to make it clearer for others to understand.